gobblin-core/src/main/java/gobblin/writer/AvroHdfsTimePartitionedWriter.java

```diff
@@ -247,16 +247,25 @@ public long bytesWritten() throws IOException {
 
   @Override
   public void close() throws IOException {
+    String fileName = WriterUtils.getWriterFileName(this.properties, this.numBranches, this.branch, this.writerId,
+        this.writerOutputFormat.getExtension());
+    List<String> pathToRecordCountMapList = Lists.newArrayList();
+
     boolean closeFailed = false;
     for (Entry<Path, DataWriter<GenericRecord>> entry : this.pathToWriterMap.entrySet()) {
       try {
         entry.getValue().close();
+        pathToRecordCountMapList.add(new Path(entry.getKey(), fileName) + ":" + entry.getValue().recordsWritten());
```

Comment:
Seems it's easier to just rename the files here, rather than passing the record count to publishers. Are there benefits of doing so?